internal/project/typesmap.go

```diff
@@ -0,0 +1,505 @@
+package project
+
+type SafeListEntry struct {
+	Match   string
+	Exclude [][]any
+	Types   []string
+}
```

Comment:
FWIW this all could be unexported, given it's only used within this package. (e.g, `safeListEntry`).